### PR TITLE
[FEAT] Testnet button to burn land expansions

### PR DESCRIPTION
**Pull request by c-py**
# Description

This PR introduces an extra button on testnet to burn land expansions. 

The code has purposely been bundled into a single file,to prevent test code from leaking all over the code base.

The code does the following:

1. Get balance of land expansions
2. Get the token IDs of the land expansions (up to the balance)
3. Use the web3 batch APIs to approve multiple burn transactions

burn

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Manual

# Checklist:

- [X] Title of the PR is relevant and is prefixed with [FEAT], [CHORE] or [FIX]
- [ ] Screenshot if it includes any UI changes
- [X] I have read the contributing guidelines and agree to the T&Cs
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/features/farming/hud/components/DEV_BurnLandButton.tsx [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+
+const loadTokenIds = async (web3: Web3, address: string, balance: number) => {
+  const tokenOfOwnerByIndex = (index: number) =>
+    web3.eth.abi.encodeFunctionCall(TOKEN_OF_OWNER_BY_INDEX_ABI, [
+      address,
+      String(index),
+    ]);
+
+  const tokenIds = await Promise.all(
+    Array.from({ length: balance }, (_, i) =>
```
Comment (spencerdezartsmith) [inline src/features/farming/hud/components/DEV_BurnLandButton.tsx:67]: 😻

File: src/features/farming/hud/components/DEV_BurnLandButton.tsx [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+  await new Promise((reject, resolve) => {
+    checkedTokens.forEach(([id]) =>
+      batch.add(
+        (web3.eth.sendTransaction as any).request(
+          {
+            from: account,
+            to: LAND_EXPANSION_ADDRESS,
+            data: web3.eth.abi.encodeFunctionCall(GAME_BURN_ABI, [id]),
+          },
+          () => ++resolved === checkedTokens.length && resolve()
```
Comment (spencerdezartsmith) [inline src/features/farming/hud/components/DEV_BurnLandButton.tsx:105]: ❤️‍🔥

Comment (c-py) [reply src/features/farming/hud/components/DEV_BurnLandButton.tsx:105]: Lol I had a bit of fun because this is test code 🤣

**Review by spencerdezartsmith — APPROVED** [review]
Watching the budding love affair with `Object.entries()` was a beautiful thing 🥲 

Tested up until game role block. I am happy to approve now.

Excellent job!! Great test feature 🏆

**Pull request merged**